employees/tests/test_unit_report_viewset.py

```diff
@@ -642,3 +643,100 @@ def test_project_report_list_get_queryset_method_should_return_queryset_containi
+        )
+        self.user.full_clean()
+        self.user.save()
+
+        self.project = Project(name="Test Project", start_date=datetime.datetime.now())
+        self.project.full_clean()
+        self.project.save()
+
+        self.report = Report(
+            date=datetime.datetime.now().date(),
+            description="Some description",
+            author=self.user,
+            project=self.project,
+            work_hours=Decimal("8.00"),
+        )
+        self.report.full_clean()
+        self.report.save()
+
+    def test_project_report_detail_view_should_display_report_details(self):
+        request = APIRequestFactory().get(path=reverse("project-report-detail", args=(self.report.pk,)))
+        request.user = self.user
+        response = ProjectReportDetail.as_view()(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.report.description)
+        self.assertEqual(response.data["serializer"].instance, self.report)
+
+    def test_project_report_detail_view_should_not_be_accessible_for_unauthenticated_users(self):
+        request = APIRequestFactory().get(path=reverse("project-report-detail", args=(self.report.pk,)))
+        request.user = AnonymousUser()
+        response = ProjectReportDetail.as_view()(request, pk=self.report.pk)
+        self.assertEqual(response.status_code, 403)
+
+    def test_project_report_detail_view_should_not_render_non_existing_report(self):
+        request = APIRequestFactory().get(path=reverse("project-report-detail", args=(999,)))
+        request.user = self.user
+        response = ProjectReportDetail.as_view()(request, pk=999)
+        self.assertEqual(response.status_code, 404)
+
+    def test_project_report_detail_view_should_update_report_on_post(self):
+        new_description = "Some other description"
+        request = APIRequestFactory().post(
+            path=reverse("project-report-detail", args=(self.report.pk,)),
+            data={
+                "date": datetime.datetime.now().date(),
+                "description": new_description,
+                "project": self.project,
+                "work_hours": Decimal("8.00"),
+            },
+        )
+        request.user = self.user
+        response = ProjectReportDetail.as_view()(request, pk=self.report.pk)
+        self.report.refresh_from_db()
+        current_description = self.report.description
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(current_description, new_description)
+        self.assertTrue(self.report.editable)
+
+    def test_project_report_detail_view_should_not_update_report_on_discard(self):
+        new_description = "Some other description"
+        request = APIRequestFactory().post(
+            path=reverse("project-report-detail", args=(self.report.pk,)),
+            data={
+                "date": datetime.datetime.now().date(),
```

Comment:
If its reused (or part of it is), add it to `setUp()` like `self.data`.

Reply:
Done